app/src/main/java/com/eveningoutpost/dexdrip/LibreAlarmReceiver.java

```diff
@@ -168,6 +171,15 @@ public void run() {
     }
 
     public static void processReadingDataTransferObject(ReadingData.TransferObject object) {
+    	Log.i(TAG, "Data that was recieved from librealarm is " + HexDump.dumpHexString(object.data.raw_data));
```

Comment:
What will happen with data from received from versions of LibreAlarm which do not support this field? Will this produce a crash or will the field not be initialized? If it is not initialized then it shouldn't be saved by `LibreBlock` (or otherwise processed) More checking should be done to ignore missing data.

Reply:
I was running with the older version, and the value that was passed for raw_data was NULL.

So, the program does not crash, and the only thing that the user will see in the log is:
LibreBlock - cannot save block with null data.
LibreOOPAlgorithm senddata called with null data.

I'll now add some code that will print in this case:
Please update librealarm to use OOP algorithm.